addons/edi/models/edi_gateway.py

```diff
@@ -193,8 +193,17 @@ def _compute_path_count(self):
     @api.depends('transfer_ids')
     def _compute_transfer_count(self):
         """Compute number of transfers (for UI display)"""
+
+        # We use the read_group method to aggregate data from the edi.transfer model. 
+        # We specify the search domain to filter transfers related to the current edi.gateway, 
+        # group the results by gateway_id, and count the number of records for each group.
+    
+        transfers_data = self.env['edi.transfer'].read_group(
+        [('gateway_id', 'in', self.ids)],
+        ['gateway_id'], ['gateway_id'])
+        transfer_count_dict = {data['gateway_id'][0]: data['gateway_id_count'] for data in transfers_data}
```

Comment:
Nit: black formatting
```py
        transfers_data = self.env["edi.transfer"].read_group(
            [("gateway_id", "in", self.ids)], ["gateway_id"], ["gateway_id"]
        )
        transfer_count_dict = {
            data["gateway_id"][0]: data["gateway_id_count"] for data in transfers_data
        }
```

Reply:
changed.

Reply:
Don't see this changed , maybe i am missing smth

Reply:
there are 2 commits can u check please.